### PR TITLE
Optimize import

### PR DESCRIPTION
**Pull request by palaviv**
I implemented CPython [PyImport_ImportModuleLevelObject](https://github.com/python/cpython/blob/59ad110d7a7784d53d0b502eebce0346597a6bef/Python/import.c#L1764) trying to improve the import speed. The problem is that it has almost no effect. For the following script:
```py
import time

start = time.time()
import os
print(time.time() - start)
```
I get after the change 0.11813092231750488 and before 0.12146615982055664.
Do you think we should add this change?

### CODE REVIEW

**Comment by windelbouwman** [comment]
I think we should import all import mechanisms in rust. Not because of performance, but for simplicity. We will not require the frozen importlib stuff, but do all in rust. I think one of the reasons that cpython uses python for the importlib, is that C is such a pain to program in. Rust is a lot better than C, so we can probably implement more in rust.

That being said, I'd rather not change the current working importlib stuff for performance, unless it improves performance. For performance, I think it is better to look at the load_name opcode, which is used a lot, and consumes a lot of time (accoring to flame-it).

File: vm/src/vm.rs [hunk 1]
```diff
@@ -311,34 +313,29 @@ impl VirtualMachine {
     }
 
     pub fn import(&self, module: &str, from_list: &PyObjectRef, level: usize) -> PyResult {
-        let sys_modules = self.get_attribute(self.sys_module.clone(), "modules")?;
-        sys_modules
```
Comment (windelbouwman) [inline vm/src/vm.rs]: Did you remove the caching mechanism (if name is in `sys_modules`) on purpose?

Comment (palaviv) [reply vm/src/vm.rs]: Yes. The first thing the `import` function does is to check for that.

**Comment by palaviv** [comment]
I am not sure I agree with changing all the import mechanism to rust but we can discuss this later. I will close this PR as we don't gain enough from this.

**Pull request closed without merging**